src/js/basicRequests.js

```diff
@@ -0,0 +1,11 @@
+function getTimeline() {
+    console.info('Requesting timeline from backend.');
+    let xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange = function(){ // more on readystates: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState
+        if(this.readyState == 4 && this.status == 200){ // waits for request to be done (readyState 4) and successful (status 200)
```

Comment:
are there any existing variables you can use for the state or status?

Reply:
I can use XMLHttpRequest.DONE for the state (added, will be pushed with the other changes) but I can't find a variable for the status. Should I get it from a different source?